proposals/automating-releases.md

```diff
@@ -0,0 +1,185 @@
+process will become much simpler, hopefully to the point where a new contributor can
+run the release with little process knowledge.
+
+## Motivation
+
+Releasing the operator-sdk repo currently requires several steps that are mostly manual,
+with some automation around building binaries for all architectures supported. Tools like
+[`goreleaser`][goreleaser] can automate this process entirely by tagging code, and building
+and publishing binaries and images with one "click".
+
+## Goals
+
+- Remove all manual steps from the release process
+- Centralize release configuration
+- One-click releasing
+
+### Non-Goals
+
+- Change the content or constituency of output release artifacts in any way
+- Change the testing process in any way
+- Change our CI provider
+
+## Proposal
+
+### User Stories
+
+#### Story 1 - Release the operator-sdk repo in one click
+
+As a contributor I want to run the release in as few operations as possible.
+This feature will enable new contributors to release the repo easily, instead
+of relying on a long onboarding document and the expertise of a few contributors.
+
+#### Story 2 - Easily modify release configuration
+
+As a contributor I want to easily improve the release configuration whenever necessary,
+which includes adding new release architectures. Configuration should be documented,
+comprehensible, and centralized.
+
+### Implementation Details/Notes/Constraints
+
+The [`goreleaser`][goreleaser] release tool satisfies the above goals and user stories:
+- Manages code tags
+- Builds and pushes [binaries][goreleaser-build] and [images][goreleaser-docker]
+  - Applies to `operator-sdk`, `ansible-operator`, and `helm-operator`
+- Centralized configuration and a well-documented [configuration spec][goreleaser-config]
+- Handles the full release process with one click
+
+`goreleaser` configuration is CI-agnostic, so if we choose to switch CI systems in the future
+we can bring our configuration with us verbatim.
+
+Additionally it supports generating release notes based on commits, or from a
+[custom source][goreleaser-custom-changelog]; we currently use a custom changelog generator
+which we can continue using.
+
+### Risks and Mitigations
+
+- Contributors need to learn a new configuration format. Luckily the goreleaser
+[config file format][goreleaser-config] is straightforward and has great documentation.
+- Release tool development is controlled by another organization, which can lead to undesirable bugs/changes.
+This is a manageable risk because the project is open source, to which we can contribute.
+
+## Design Details
+
+### Test Plan
+
+1. Create a goreleaser configuration that creates the exact same release artifacts the current process does.
+1. Push configuration to a test repository.
+1. Run a test release.
+1. Ensure:
+  1. Tag is present remotely.
+  1. Release notes/binaries are published as expected.
+  1. Images are pullable and run on the desired architectures.
```

Comment:
Indent these one more space so that the subbullets line up with the E in Ensure. It doesn't render correctly as it written.
![image](https://user-images.githubusercontent.com/17137/94502988-6211e800-01d3-11eb-9370-21c1b0cb6075.png)